hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/request/key/TestOMDirectoriesPurgeRequestAndResponse.java

```diff
@@ -225,4 +225,73 @@ public void testValidateAndUpdateCacheCheckQuota() throws Exception {
           deletedKey));
     }
   }
+
+  @Test
+  public void testValidateAndUpdateCacheQuotaBucketRecreated()
+      throws Exception {
+    // Create and Delete keys. The keys should be moved to DeletedKeys table
+    List<OmKeyInfo> deletedKeyInfos = createAndDeleteKeys(1, null);
+    // The keys should be present in the DeletedKeys table before purging
+    List<String> deletedKeyNames = new ArrayList<>();
+    for (OmKeyInfo deletedKey : deletedKeyInfos) {
+      String keyName = omMetadataManager.getOzoneKey(deletedKey.getVolumeName(),
+          deletedKey.getBucketName(), deletedKey.getKeyName());
+      Assert.assertTrue(omMetadataManager.getDeletedTable().isExist(
+          keyName));
+      deletedKeyNames.add(keyName);
+    }
+
```

Comment:
Thanks @sumitagrawl changes look great.  One thing on the tests within this set of tests - code reuse.  These two tests, `testValidateAndUpdateCacheQuotaBucketRecreated `and `testValidateAndUpdateCacheCheckQuota `both use blocks of code that can be reused.  It would be great have block of code that checks the `deletedKeyInfos` are in the `deletedTable` and returns the `deletedKeyNames,` something like 
`private List<String> validateDeletedKeysTable(List<OmKeyInfo> deletedKeyInfos)`
and is used in both tests in this unit test file.

Reply:
Updated... @neils-dev